src/main/java/no/nav/fpsak/tidsserie/LocalDateTimeline.java

```diff
@@ -722,11 +722,11 @@ private void handleGapInExistingTimeline(LocalDateSegment<V> datoSegment, LocalD
 
         if (gapBak != null) {
             Optional<LocalDateInterval> overlap = datoInterval.overlap(gapBak);
-            overlap.ifPresent(di -> newSegments.add(new LocalDateSegment<V>(di, datoSegment.getValue())));
+            overlap.ifPresent(di -> newSegments.add(segmentSplitter.apply(di, datoSegment)));
```

Comment:
Bør du gjøre noe med default - som er EqualValueSegmentSplitter som igjen gjør new LocalDateSegment(di, seg.value()) ? 
Eller holder det å forutsette at man supplyer en custom segmentsplitter? 
Det er generelt to tilfelle som er tricky - Containers (Collection/Map) og objekt med aggregerte verdier (anti-pattern) - fx beløp = dagsats * dager i periode

Reply:
@jolarsen Litt usikker, egentlig. Har ikke full oversikt over alle steder den brukes, så ville ikke rokke for mye med den. Men `createNewListOnSplit` vil vel kun funke om `V` er en collection. Det smudeste hadde kanskje vært å velge `createNewListOnSplit` som default dersom `V` er en collection, og ellers `EqualValueSegmentSplitter`, men så vidt jeg vet er ikke det mulig.